File: test-suite/src/test/java/io/micronaut/kafka/docs/quickstart/QuickstartTest.java
package io.micronaut.kafka.docs.quickstart;

import io.micronaut.context.ApplicationContext;
import io.micronaut.context.annotation.Property;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.micronaut.test.support.TestPropertyProvider;
import io.micronaut.testcontainers.kafka.Kafka;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import java.util.Map;

@Property(name = "spec.name", value = "QuickstartTest")
@Property(name = "kafka.enabled", value = "true")
@MicronautTest
class QuickstartTest implements TestPropertyProvider {
    @Inject
    ApplicationContext applicationContext;

    @Override
    public Map<String, String> getProperties() {
        return Kafka.getProperties();
    }

    @Test
    void testSendProduct() {
        // tag::quickstart[]
        ProductClient client = applicationContext.getBean(ProductClient.class);
        client.sendProduct("Nike", "Blue Trainers");
        // end::quickstart[]
    }
}

File: test-suite-python/src/test/python/micronaut/kafka/docs/quickstart/QuickstartTest.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context import ApplicationContext
from micronaut.context.annotation import Property
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from .ProductClient import ProductClient


@Property(name="spec.name", value="QuickstartTest")
@Property(name="kafka.enabled", value="true")
@MicronautTest(environments=["kafka"])
class QuickstartTest:
    applicationContext: Annotated[ApplicationContext, Inject]

    @Test
    def test_send_product(self):
        # tag::quickstart[]
        client = self.applicationContext.getBean(ProductClient).asPolyglotValue()
        client.send_product("Nike", "Blue Trainers")
        # end::quickstart[]
